### ISSUE TITLE
show out of stock tag should not be optional if stock management is disabled

### ISSUE THREAD

**Issue by alfonsolr09** [PrestaShop/PrestaShop#28098]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

first I think that the option with the red rectangle should be placed under the stock management and not at the bottom

Next, I think that if stock management is disabled, the "show out of stock label" option should not be able to be modified in the same way as the "allow sale of products that are out of stock" option does.

![image](https://user-images.githubusercontent.com/62474824/161162471-0676ad4c-78f3-4c54-b6b7-50371d863a28.png)

When 'Allow ordering of out-of-stock products'  is set to 'No'
Then the toggle of 'Display out-of-stock label on product listing pages' is disabled

When 'Allow ordering of out-of-stock products'  is set to 'Yes'
Then the toggle of 'Display out-of-stock label on product listing pages' is enabled

### PrestaShop version(s) where the bug happened

1.7.8.5

### PHP version(s) where the bug happened

7.3.33

**Comment by khouloudbelguith** [comment]
Hi @alfonsolr09,

Thanks for your report.
In fact, we have a feature ready and validated by the product team about the “out of stock flag on product listings and product pages + the toggle in the BO product settings page as described here: https://github.com/PrestaShop/PrestaShop/issues/24759
It is a related to this epic: PrestaShop/hummingbird#280

Could you please check if it is similar to your issue and feedback!

**Comment by Amit-Kumar-Tiwari-Webkul** [comment]
@khouloudbelguith , Yes I agree but currently we can fix it in 1.7.8.6 patch PR is already available, it seems this button is going to be removed in 8.0.0

**Comment by Amit-Kumar-Tiwari-Webkul** [comment]
HI,
Any update?

**Comment by hibatallahAouadni** [comment]
Hello @Amit-Kumar-Tiwari-Webkul & @alfonsolr09 

What you are asking is a feature (small one) to put this option **Display out-of-stock label on product listing pages** disbaled by default as the **Enable stock management** option.
So, let's check with Product Team :wink: 
Ping @PrestaShop/product-team do you think that we're able to add this small feature to the 1.7.8 releases?

Thanks!

**Comment by alfonsolr09** [comment]
> Hi @alfonsolr09,
> 
> Thanks for your report. In fact, we have a feature ready and validated by the product team about the “out of stock flag on product listings and product pages + the toggle in the BO product settings page as described here: PrestaShop/PrestaShop#24759 It is a related to this epic: PrestaShop/hummingbird#280
> 
> Could you please check if it is similar to your issue and feedback!

I think they are different problems
I suggest the toggle switch be disabled if stock management is disabled

**Comment by alfonsolr09** [comment]
I see a little confusion so I am about to clarify the issue. we start from the basis that there are 4 possible changes/improvements.

If "stock management" is disabled, the "show out of stock label" option should not be able to be modified (set on NO) in the same way as the "allow sale of products that are out of stock" option does (set on YES).

Stock management is enabled.- When 'Allow ordering of out-of-stock products' is set to 'No'
Then the toggle of 'Display out-of-stock label on product listing pages' is disabled (set on YES)

When 'Allow ordering of out-of-stock products' is set to 'Yes'
Then the toggle of 'Display out-of-stock label on product listing pages' is enabled (set by default on YES)

I propose that all the toggles switch be one on top of the other (or all grouped in the same position) to be able to make the configurations without having to scroll from top to bottom
